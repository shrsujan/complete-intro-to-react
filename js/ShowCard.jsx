import React from 'react';
import PropTypes from 'prop-types';

class ShowCard extends React.Component {
  render() {
    return (
      <div className="show-card">
        <img alt={`${this.props.show.title} Show Poster`} src={`/public/img/posters/${this.props.show.poster}`} />
        <div>
          <h3>{this.props.show.title}</h3>
          <h4>({this.props.show.year})</h4>
          <p>{this.props.show.description}</p>
        </div>
      </div>
    );
  }
}

ShowCard.propTypes = {
  show: PropTypes.object
};

export default ShowCard;
