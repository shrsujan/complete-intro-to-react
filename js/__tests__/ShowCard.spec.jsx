// @flow

import React from 'react';
import { shallow } from 'enzyme';
import ShowCard from '../ShowCard';

describe('ShowCard', () => {
  it('should render as per given props', () => {
    const fakeProps = {
      title: 'Atlanta',
      year: '2008â€“2013',
      description: 'Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest \'Earn\' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.',
      poster: 'a.jpg',
      imdbID: 'tt4288182',
      trailer: 'MpEdJ-mmTlY'
    };
    const component = shallow(<ShowCard key={fakeProps.imdbID} {...fakeProps} />);
    expect(component).toMatchSnapshot();
  });
});
