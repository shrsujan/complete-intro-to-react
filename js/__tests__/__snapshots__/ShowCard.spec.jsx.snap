// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCard should render as per given props 1`] = `
<styled.div>
  <styled.img
    alt="Atlanta Show Poster"
    src="/public/img/posters/a.jpg"
  />
  <div>
    <h3>
      Atlanta
    </h3>
    <h4>
      (
      2008â€“2013
      )
    </h4>
    <p>
      Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest 'Earn' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.
    </p>
  </div>
</styled.div>
`;
